File: ChessOpenings/ChessOpenings/Data/PgnReader.java
import java.io.*;
import java.util.*;

public class PgnReader
{
	public static final String PATH = "D:/Dev/Data Files/Chess/";
	public static final String FILENAME = "ficsgamesdb_201801_standard2000_nomovetimes_1543578.pgn";
	public BufferedReader br = null;
	public FileReader fr = null;
	public List<Game> games = new ArrayList<Game>();


	public List<Game> getGamesList()
	{
		try
		{
			fr = new FileReader(PATH + FILENAME);
			br = new BufferedReader(fr);
		
		
			String currentGame = "";

			String currentLine = br.readLine();
			while (currentLine != null)
			{
				currentGame += currentLine + "\n";
				if (currentLine.endsWith("1/2-1/2") || currentLine.endsWith("1-0") || currentLine.endsWith("0-1"))
				{
					games.add(new Game(currentGame));
					//currentGame = trimGame(currentGame);
					//games.add(currentGame);
					currentGame = "";
				}
				currentLine = br.readLine();
			}
		}
		catch (Exception e)
		{
			e.printStackTrace();
		}
		for (Game g : games)
		{
			System.out.println("game :- " + g.toString());
		}
		return games;
	}

	private String trimGame(String game)
	{
		String rtn;
		rtn = game.replaceAll("\\[.*\\]", "");
		rtn = game.replaceAll("{.*}", "");
		rtn = game.replaceAll("\\$\\d", "");
		rtn = game.replaceAll("\\d*\\.", "");
		return rtn;
	}

}

File: ChessOpenings.DataConversion/PgnReader.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChessOpenings.DataConversion
{
    public class PgnReader
    {
        public static string PATH = "E:/";
        public static string FILENAME = "ficsgamesdb_201801_standard2000_nomovetimes_1543578.pgn";
        public StreamReader reader;
        public List<Game> games = new List<Game>();

        public List<Game> GetGamesList()
        {
            reader = new StreamReader(PATH + FILENAME);

            string currentGame = "";

            string currentLine = reader.ReadLine();
            while (currentLine != null)
            {
                currentGame += currentLine + "\n";
                if (currentLine.EndsWith("1/2-1/2") || currentLine.EndsWith("1-0") || currentLine.EndsWith("0-1"))
                {
                    games.Add(new Game(currentGame));
                    currentGame = "";
                }
                currentLine = reader.ReadLine();
            }

            return games;
        }
    }
}
